catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/forms/model/JsonModelBuilderTest.java

```diff
@@ -0,0 +1,214 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.forms.model;
+
+import static org.codice.ddf.catalog.ui.forms.model.FilterNodeAssertionSupport.assertLeafNode;
+import static org.codice.ddf.catalog.ui.forms.model.FilterNodeAssertionSupport.assertParentNode;
+import static org.codice.ddf.catalog.ui.forms.model.FilterNodeAssertionSupport.assertTemplatedNode;
+
+import org.codice.ddf.catalog.ui.forms.model.pojo.FilterNode;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.runners.Enclosed;
+import org.junit.runner.RunWith;
+
+@RunWith(Enclosed.class)
+public class JsonModelBuilderTest {
+  private static final String AND = "AND";
+
+  private static final String OR = "OR";
+
+  private static void setupDefaultTestValue(JsonModelBuilder builder) {
+    builder
+        .beginBinaryComparisonType("=")
+        .setProperty("name")
+        .setValue("value")
+        .endTerminalType()
+        .getResult();
+  }
+
+  public static class ValidateCorrectUse {
+    private JsonModelBuilder builder;
+
+    @Before
+    public void setup() {
+      builder = new JsonModelBuilder();
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void testBeginBinaryLogicTypeCanModify() {
```

Comment:
✏️ I understand why the tests are named this way but it's still confusing. The way this test is named (and in the same way most of the tests in this class are named), I would expect this test to verify that `BeginBinaryLogicType` is able to modify something. It's not clear what `CanModify` means just by inspecting the test. What if the test names followed a pattern like `beginBinaryLogicTypeAfterResultObtainedThrowsIllegalStateException`?

Reply:
I understand the confusion. But I think if we add too much the names will get too difficult to manage. In this case there is a clear mapping between the method under test and the validation method being called behind the scenes. 

To that end, I've added a comment to help clarify things while keeping the methods as concise as they are now.